test/acceptance/user.controller.acceptance.ts

```diff
@@ -109,23 +97,27 @@ describe('UserController', () => {
   it('returns a user with given id when GET /user/{id} is invoked', async () => {
     const newUser = await userRepo.create(user);
     delete newUser.password;
+    delete newUser.orders;
     // MongoDB returns an id object we need to convert to string
     // since the REST API returns a string for the id property.
     newUser.id = newUser.id.toString();
 
     await client.get(`/users/${newUser.id}`).expect(200, newUser.toJSON());
   });
 
-  function givenAnApplication() {
+  async function setupApplication() {
```

Comment:
`setupApplication` is duplicated in acceptance tests. Can we extract a shared helper please?

I understand `setupApplication` is updating global variables. Can we leverage object destructuring as an easy-to-use solution?

```ts
// the helper
async function setupApplication() {
  // ...
  return {app, server, client};
}


// usage in tests
let app: Application;
let server: RestServer;
let client: supertest.SuperTest<supertest.Test>;

before(async () => {
  // notice no "const", "let" or "var"
  {app, server, client} = await setupApplication();
});
```

Also since the function is not setting only the app, but also the server and the client, I think we should use a different name, e.g. `setupAcceptanceWorkbench`.